ppa/archive/gale.py

```diff
@@ -13,32 +14,36 @@
 logger = logging.getLogger(__name__)
 
 
-def get_local_ocr(item_id, page_num):
+def get_local_ocr(item_id):
     """
-    Get local OCR page text for specified page of a single Gale volume if available.
+    Load local OCR page text for the specified Gale volume, if available.
     This requires a base directory (specified by GALE_LOCAL_OCR) to be configured and
     assumes the following organization:
-        
+
         * Volume-level directories are organized in stub directories that correspond to
           every third number (e.g., CW0128905397 --> 193). So, a Gale volume's OCR data
-          is located in the following directory: GALE_LOCAL_OCR / stub_dir / item_id
-        
-        * A page's local ocr text file is named [volume id]_[page number]0.txt, where the
-          page number is a 4-digit string (e.g., "0004"). So, for page number "0004" in
-          volume CW0128905397, the local ocr file is named "CW0128905397_00040.txt".
+          is located in the following directory: GALE_LOCAL_OCR / stub_dir / item_id.json
+
+        * Page text is stored as a JSON dictionary with keys based on Gale page numbers,
+          which is a 4-digit string (e.g., "0004").
 
-    Raises a FileNotFoundError if the local OCR page text does not exist. 
+    Raises a FileNotFoundError if the local OCR page text does not exist.
     """
     ocr_dir = getattr(settings, "GALE_LOCAL_OCR", None)
     if not ocr_dir:
         raise ImproperlyConfigured(
             "GALE_LOCAL_OCR configuration is required for indexing Gale page content"
         )
+    # check that the id looks as expected (appease github codeql security concerns)
+    # first two characters are CW or CB; rest of the id is numeric
+    if not all([item_id[:2] in ["CW", "CB"], item_id[2:].isnumeric()]):
+        raise ValueError(f"{item_id} is not a valid Gale item identifier")
 
-    stub_dir = item_id[::3][1:]  # Following conventions set in ppa-nlp
-    ocr_txt_fp = os.path.join(ocr_dir, stub_dir, item_id, f"{item_id}_{page_num}0.txt")
-    with open(ocr_txt_fp) as reader:
-        return reader.read()
+    # files are in stub directories; following conventions set in ppa-nlp
+    stub_dir = item_id[::3][1:]
+    ocr_dir = pathlib.Path(ocr_dir)
+    with (ocr_dir / stub_dir / f"{item_id}.json").open() as ocrfile:
```

Comment:
Why not just make the full path for the ocrfile?
```suggestion
    ocr_path = pathlib.Path(ocr_dir, stub_dir, f"{item_id}.json")
    with ocr_path.open() as ocrfile:
```